programs/aes/crypt_and_hash.c

```diff
@@ -103,7 +103,14 @@ int main(int argc, char *argv[])
         list = mbedtls_cipher_list();
         while (*list) {
             cipher_info = mbedtls_cipher_info_from_type(*list);
-            mbedtls_printf("  %s\n", mbedtls_cipher_info_get_name(cipher_info));
+            if (cipher_info) {
+                const char *name = mbedtls_cipher_info_get_name(cipher_info);
+
+                if (name) {
+                    mbedtls_printf("  %s\n", mbedtls_cipher_info_get_name(cipher_info));
+                }
+            }
+
```

Comment:
Why not do this in a way that gives some output, which might let the user (which could be us) know that something is wrong? e.g.

```
            const char *name = (cipher_info == NULL) ? NULL : mbedtls_cipher_info_get_name(cipher_info);
            mbedtls_printf("  %s\n", (name == NULL) ? "(missing)" : name);
```

Reply:
This is purely just keeping the compiler happy with various sanitisers, its not actually a correctness issue.